presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -156,12 +156,19 @@ public void finish()
 
         checkSuccess(spillInProgress, "spilling failed");
         finishing = true;
+
+        // check if fully finished (and close everything if so)
+        isFinished();
     }
 
     @Override
     public boolean isFinished()
     {
-        boolean finished = this.finished && probe == null && pageBuilder.isEmpty() && outputPage == null;
+        if (finished) {
+            return true;
+        }
+
+        boolean finished = finishing && probe == null && pageBuilder.isEmpty() && outputPage == null;
```

Comment:
@findepi the change on this line from `this.finished` to `finishing`, I think is a fix to the change introduced in https://github.com/prestodb/presto/commit/95ad82b47530f8ecef59bd1afc215894965e2acd#diff-87982652850e9429e242e7ddf08e77fdR162  Does this look right to you?

Reply:
I'm pretty sure that this breaks when the probe is spilled, and this only checks what is loaded into memory.  Is there any way to know if there is any spilled probe data.

Reply:
You can check if there was anything spilled with `spiller.isPresent` (however, now, once this is true, it will always be). Let me think about that, ok?

Reply:
> I think is a fix to the change introduced in https://github.com/prestodb/presto/commit/95ad82b47530f8ecef59bd1afc215894965e2acd#diff-87982652850e9429e242e7ddf08e77fdR162

Could you explain what was wrong? 
(i think `this.finished` was becoming `true` only after we have unspilled everything)

Reply:
If you do a join with an empty probe the operator waits until the build finishes, which could take hours.

Reply:
Oh ya, the variable name is just confusing.... the code says we're finished if we are finished and there is no more input or output.

Reply:
> If you do a join with an empty probe the operator waits until the build finishes, which could take hours.

i was aware of this limitation and we've been discussing this with @losipiuk last summer. Question is  -- how much of a problem is this? 
Ideally, we should be quite good at knowing which side of the join is bigger, so empty probe side and hours-worth of the build side is not something we can be proud of.